python/TestHarness/schedulers/Scheduler.py

```diff
@@ -136,48 +136,27 @@ def processDownstreamTests(self, job_container):
 
         return failed_job_containers
 
-    def createDependencies(self, job_container_dict):
-        """
-        Create dependencies using supplied dictionary of TesterData objects.
-        """
-        for tester_name, job_container in job_container_dict.iteritems():
-            tester = job_container.getTester()
-            job_dag = job_container.getDAG()
-
-            job_dag.add_node_if_not_exists(job_container)
-
-            for prereq in tester.getPrereqs():
-
-                if prereq in job_container_dict.keys():
-
-                    job_dag.add_node_if_not_exists(job_container_dict[prereq])
-
-                    # Attempt to create an edge node dependency
-                    try:
-                        job_dag.add_edge(job_container_dict[prereq], job_container)
-
-                    # We do not care about cyclic errors, they have been handled already
-                    except dag.DAGValidationError:
-                        pass
-
-        return job_container_dict
-
     def checkGroupFailures(self, job_container_dict):
         """
-        Method to detect failures on a macroscopic level. Returns a set of tests
-        that were discovered as causing failures.
+        Method to detect failures on a macroscopic level. Returns a set of testers
+        that are discovered to cause failures.
         """
-        temp_dag = dag.DAG()
+
         failed_or_skipped_testers = set([])
 
+        # Arbitrarily obtain a dag object from one of the job containers as they
+        # are shared between each job_container
+        arbitrary_job = job_container_dict.iteritems().next()[1]
```

Comment:
A job_container (TesterData), which contains the shared DAG we need to work with